Maple2.Server.Game/Model/Field/Actor/ActorState/AiState.cs

```diff
@@ -0,0 +1,452 @@
+﻿using Maple2.Model.Enum;
+using Maple2.Model.Metadata;
+using Maple2.Server.Core.Constants;
+using Maple2.Server.Game.Model.Skill;
+using Maple2.Server.Game.Packets;
+using Microsoft.EntityFrameworkCore.Metadata.Conventions;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Numerics;
+using System.Text;
+using System.Threading.Tasks;
+using static Maple2.Model.Metadata.AiMetadata;
+using static System.Runtime.InteropServices.JavaScript.JSType;
+
+namespace Maple2.Server.Game.Model.Field.Actor.ActorState;
+
+public class AiState {
+    private readonly FieldNpc actor;
+    public AiMetadata? AiMetadata { get; private set; }
+    private Node? battle = null;
+
+    private List<StackEntry> aiStack = new List<StackEntry>();
+
+    public AiState(FieldNpc actor) {
+        this.actor = actor;
+    }
+
+    [MemberNotNullWhen(true, "AiMetadata")]
+    public bool SetAi(string name) {
+        if (name == string.Empty) {
+            AiMetadata = null;
+
+            return false;
+        }
+
+        AiMetadata? metadata;
+
+        if (!actor.Field.AiMetadata.TryGet(name, out metadata)) {
+            return false;
+        }
+
+        AiMetadata = metadata;
+
+        battle = new Node("battle", AiMetadata.Battle);
+
+        return true;
+    }
```

Comment:
Didn't we determine that the AI is set once and never changed? We should just set it in the constructor and change the AiMetadata to readonly.
Summoned pets would be different, as they'll always be summoned as FieldPet rather than FieldNpc. So that means we can set the AI path in the constructor there too.

Reply:
I agree that it should eventually be done, but I'd rather wait until we start testing with working implementations of AI in the wild & check some of the hairier edge cases in the game before making that judgement call. This is a fairly easy change to do, but doing it too early could make it harder to undo cause of difference in nullability

Reply:
sounds good just drop a todo so we remember to go back when the time is right